packages/jest/src/jest-configuration-builder.ts

```diff
@@ -1,8 +1,38 @@
 import { Path, resolve } from "@angular-devkit/core";
-import { merge } from 'lodash';
+import { mergeWith } from 'lodash';
 import { CustomConfigResolver } from "./custom-config.resolver";
 import { DefaultConfigResolver } from "./default-config.resolver";
 
+/**
+ * A whitelist of property names that are mean to be concat.
+ */
+const ARRAY_PROPERTIES_TO_CONCAT = [
+  // From Jest Config
+  'setupFilesAfterEnv',
+  // From ts-jest config
+  'astTransformers'
+];
+
+/**
+ * This function checks witch properties should be concat. Early return will
+ * merge the data as lodash#merge would do it. 
+ */
+function concatArrayProperties(objValue: unknown, srcValue: unknown, property: string) {
+  if (!ARRAY_PROPERTIES_TO_CONCAT.includes(property)) {
+    return;
+  }
+
+  if (!Array.isArray(objValue)) {
```

Comment:
Can be united with the previous if statement

Reply:
Sure. I miss that.